### ISSUE TITLE
Trying to run on CPU :  assert outputs.is_cuda or outputs.device.type == 'xla' AssertionError

### ISSUE THREAD

**Issue by azh4r** [karpathy/nanoGPT#317]
Hi, when I try to run on CPU using:

`python train.py config/train_shakespeare_char.py --device=cpu --compile=False --eval_iters=20 --log_interval=1 --block_size=64 --batch_size=12 --n_layer=4 --n_head=4 --n_embd=128 --max_iters=2000 --lr_decay_iters=2000 --dropout=0.0`

Getting error:

  `File "/home/projects/nanoGPT/env-nano-gpt/lib/python3.10/site-packages/torch/cuda/amp/grad_scaler.py", line 166, in scale
    assert outputs.is_cuda or outputs.device.type == 'xla'
AssertionError`

**Comment by sameernyaupane** [comment]
Same thing happened to me, the solution was to uninstall pytorch with cuda enabled, and install pytorch cpu only.

``` pip uninstall torch ```

Install the nightly version for better speeds as mentioned in docs.

``` pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cpu
```
